keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/DeliveryConfigController.kt

```diff
@@ -60,16 +67,37 @@ class DeliveryConfigController(
   fun upsert(
     @RequestBody
     @SwaggerRequestBody(
-      description = "The delivery config. If its `name` matches an existing delivery config the operation is an update, otherwise a new delivery config is created."
+      description = "The delivery config. If its `name` matches an existing delivery config the operation is " +
+        "an update, otherwise a new delivery config is created."
     )
     deliveryConfig: SubmittedDeliveryConfig
-  ): DeliveryConfig =
-    deliveryConfigProcessors.applyAll(deliveryConfig)
+  ): DeliveryConfig {
+    val metadata: MutableMap<String,Any?> = deliveryConfig.metadata?.toMutableMap() ?: mutableMapOf()
+    val gitMetadata: GitMetadata? = try {
+      val candidateMetadata = metadata.getOrDefault("gitMetadata", null)
+      if (candidateMetadata != null) {
+        mapper.convertValue<GitMetadata>(candidateMetadata)
+      } else {
+        null
+      }
+    } catch (e: IllegalArgumentException) {
+      log.debug("Error converting git metadata ${metadata.getOrDefault("gitMetadata", null)}: {}", e)
+      // not properly formed, so ignore the metadata and move on
+      null
+    }
+
+    // we don't save this key to the database, since we're programmatically adding it in orca.
+    metadata.remove("gitMetadata")
```

Comment:
Why remove? We already add a ton of metadata everywhere that is returned by the API, so if you call `GET /applications/{application}/config` what you get is much different from what's in your git repo already. I propose we keep it as it can be useful for troubleshooting.

Reply:
Okay, I'll keep it. I wasn't sure what the best move was, but you're right it would be useful for troubleshooting.